### ISSUE TITLE
Element {}item invalid at this location#0

### ISSUE THREAD

**Issue by robbiemoore** [davispeixoto/Laravel-5-Salesforce#19]
Hi, like the library - thanks.

I've followed the steps, and am trying the initial `Salesforce::describeLayout('Account')`

However I get the below error message. I've donw a bit of debugging and found that the login details were verified, and a valid sessionID returned.

Line 767 where the failure occurs is this: `return $this->sforce->describeLayout($arg)->result;`

Any ideas?

`Element {}item invalid at this location#0 /home/vagrant/Code/voiceiq-api/vendor/davispeixoto/force-dot-com-toolkit-for-php/src/Davispeixoto/ForceDotComToolkitForPhp/SforceBaseClient.php(767): SoapClient->__call('describeLayout', Array) #1 /home/vagrant/Code/voiceiq-api/vendor/davispeixoto/force-dot-com-toolkit-for-php/src/Davispeixoto/ForceDotComToolkitForPhp/SforceBaseClient.php(767): SoapClient->describeLayout(Object(stdClass)) #2 [internal function]: Davispeixoto\ForceDotComToolkitForPhp\SforceBaseClient->describeLayout('Account') #3 /home/vagrant/Code/voiceiq-api/vendor/davispeixoto/laravel5-salesforce/src/Davispeixoto/Laravel5Salesforce/Salesforce.php(35): call_user_func_array(Array, Array) #4 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(237): Davispeixoto\Laravel5Salesforce\Salesforce->__call('describeLayout', Array) #5 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(237): Davispeixoto\Laravel5Salesforce\Salesforce->describeLayout('Account') #6 /home/vagrant/Code/voiceiq-api/routes/web.php(12): Illuminate\Support\Facades\Facade::__callStatic('describeLayout', Array) #7 /home/vagrant/Code/voiceiq-api/routes/web.php(12): Davispeixoto\Laravel5Salesforce\SalesforceFacade::describeLayout('Account') #8 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Route.php(176): App\Providers\RouteServiceProvider->{closure}() #9 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Route.php(147): Illuminate\Routing\Route->runCallable() #10 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Router.php(642): Illuminate\Routing\Route->run(Object(Illuminate\Http\Request)) #11 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request)) #12 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(41): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request)) #13 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(137): Illuminate\Routing\Middleware\SubstituteBindings->handle(Object(Illuminate\Http\Request), Object(Closure)) #14 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request)) #15 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(65): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request)) #16 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(137): Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle(Object(Illuminate\Http\Request), Object(Closure)) #17 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request)) #18 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request)) #19 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(137): Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure)) #20 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request)) #21 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request)) #22 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(137): Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure)) #23 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request)) #24 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request)) #25 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(137): Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure)) #26 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request)) #27 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(59): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request)) #28 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(137): Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure)) #29 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request)) #30 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(104): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request)) #31 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Router.php(644): Illuminate\Pipeline\Pipeline->then(Object(Closure)) #32 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Router.php(618): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request)) #33 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Router.php(596): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request)) #34 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(267): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request)) #35 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request)) #36 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php(46): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request)) #37 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(137): Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure)) #38 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request)) #39 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(104): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request)) #40 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(149): Illuminate\Pipeline\Pipeline->then(Object(Closure)) #41 /home/vagrant/Code/voiceiq-api/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(116): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request)) #42 /home/vagrant/Code/voiceiq-api/public/index.php(54): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request)) #43 {main}`

**Comment by davispeixoto** [comment]
@frostfat , I think this link might help you out: [https://developer.salesforce.com/forums/?id=906F00000009FdQIAU](https://developer.salesforce.com/forums/?id=906F00000009FdQIAU)

Also note that by default the wsdl files are cached on PHP library, so, if you have recently update the file, you need to verify this caching configuration and perhaps clear it up. This other link can help you with this: [http://php.net/manual/en/soap.configuration.php](http://php.net/manual/en/soap.configuration.php)

Keep us updated.

**Comment by robbiemoore** [comment]
@davispeixoto, thanks for the links! Turns out I generated a Partner wsdl, and the namespace in this package seems to default to urn:enterprise.soap.sforce.com, without detecting the wsdl type.

Turns out I need the enterprise wsdl anyway, so my solution was to generate and upload an enterprise wsdl, rather than a partner one. A better fix have been to assign the namespace depending on the wsdl type, but unnecessary for me. Thanks for your direction!
